src/main/java/ch/njol/skript/expressions/ExprBlockSound.java

```diff
@@ -0,0 +1,125 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.Sound;
+import org.bukkit.SoundGroup;
+import org.bukkit.block.Block;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+
+@Name("Block Sound")
+@Description("Gets the sound that a given block, blockdata, or itemtype will use in a specific scenario.")
```

Comment:
It would be nice to have a note about what the sound 'is' (e.g. that it's a text, looking like `BLAH_BLAH`) because I can see people being confused by that.

Maybe a wiki link to sounds would help too, so people know what to expect?